src/jdk.compiler/share/classes/com/sun/tools/javac/tree/DCTree.java

```diff
@@ -102,12 +242,64 @@ public String toString() {
         return s.toString();
     }
 
+    /**
+     * {@return the last (right-most) child of this node}
+     */
+    private DCTree getLastChild() {
+        final DCTree[] last = new DCTree[] {null};
+
+        accept(new DocTreeScanner<Void, Void>() {
+            @Override @DefinedBy(Api.COMPILER_TREE)
+            public Void scan(DocTree node, Void p) {
+                if (node instanceof DCTree dcTree) last[0] = dcTree;
+                return null;
+            }
+        }, null);
+
+        return last[0];
+    }
+
+    /**
+     * {@return a diagnostic position based on the positions in a comment}
+     *
+     * The positions are lazily converted to file-based positions as needed.
```

Comment:
"lazily" somewhat duplicates "as needed", no?

Reply:
Noted

Reply:
It doesn't read as well without "lazily" or "as needed".  Would prefer to mostly leave alone.